mlflow/tracking/client.py

```diff
@@ -2394,7 +2394,9 @@ def log_image(
 
             # Sanitize key to use in filename (replace / with # to avoid subdirectories)
             sanitized_key = re.sub(r"/", "#", key)
-            filename_uuid = uuid.uuid4()
```

Comment:
In https://github.com/mlflow/mlflow/actions/runs/12923767641/job/36041606794?pr=14264, `uuid.uuid4()` coincidentally returns a string starting with `28`, which caused the test to fail.